middleware.ts

```diff
@@ -83,7 +84,8 @@ export async function middleware(req: NextRequest) {
       logMessage.debug(
         `Middleware - Redirecting to cookie language: ${cookieLang}, pathname: ${pathname}`
       );
-      return NextResponse.redirect(new URL(`/${cookieLang}${pathname}`, req.url));
+
+      return NextResponse.redirect(new URL(`/${cookieLang}${pathname}?${searchParams}`, req.url));
```

Comment:
Minor nit pick to only include the question mark if any searchParams exist. Something like
```
`/${cookieLang}${pathname}${searchParams ?  "?"+searchParams : ""}`
```